lib/broccoli/ember-app.js

```diff
@@ -1341,15 +1321,17 @@ EmberApp.prototype.styles = function() {
     }.bind(this));
   }
 
-  this.import('vendor/addons.css');
-
   var vendorStyles = [];
   for (var outputFile in this._styleOutputFiles) {
+    var isMainVendorFile = outputFile === this.options.outputPaths.vendor.css;
     var headerFiles = this._styleOutputFiles[outputFile];
+    var inputFiles = isMainVendorFile ? ['modules/**/*.css'] : [];
```

Comment:
1. [Shows that we have a `modules` folder containing JS.](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L931)
2. [Pulls assets from _anywhere_ inside of the `treeForAddon` results.](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L937), not just the `modules` folder (note [which tree we feed it](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L925)).

It can do this because it is invoked before being combined with the `externalTree`. I _believe_ that your `concat` behavior is thus still broken. (I should actually confirm this rather than just statically analyzing the code but I'm on a plane and that's a pain in the ass to get a setup for since I've set up for my other PR.)